oaknational/Oak-Web-Application
# spike: page with css-modules

## Description

Relates #38 

- adds style-lint
- trials css-modules
- adds [clsx](https://github.com/lukeed/clsx) for classNames
- adds [css reset](https://piccalil.li/blog/a-modern-css-reset/)

### How can I test

Check the branch deployment, navigate to `/lessons/[slug]` where slug can be anything

![image](https://user-images.githubusercontent.com/12934669/157735556-99f32616-fa3d-4120-8553-b23bba320e07.png)


## Findings

So far I'm unconvinced by css-modules. Generally the vibe feels quite 2013/2014. I'm looking for best practises and finding myself trawling through conversations with the author (who later co-authored styled components) on gitter(!). I didn't add "scss" support yet, though it we were to go down the css-modules route i think we ought to. Though i think the main benefit would be that it allows us to next `@media` queries in selectors, and the use of mixins, extends, to introduce some extra logic in to the styles.

### Positives
- hot-reloading seems a lot faster (than SC)

### Negatives
- Importing between css files feels clunky, there's no typesafety or auto-complete.
- Haven't been able to get typesafe imports from css modules into js files (`typescript-plugin-css-modules` should have sorted that i thought?)
